test/hotspot/jtreg/runtime/cds/appcds/dynamicArchive/test-classes/DuplicatedCustomApp.java

```diff
@@ -0,0 +1,110 @@
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+import java.io.*;
+import java.net.*;
+import sun.hotspot.WhiteBox;
+
+public class DuplicatedCustomApp {
+    static WhiteBox wb = WhiteBox.getWhiteBox();
+    static URLClassLoader loaders[];
+
+    // If DuplicatedCustomApp.class is loaded from JAR file, it means we are dumping the
+    // dynamic archive.
+    static boolean is_dynamic_dumping = !wb.isSharedClass(DuplicatedCustomApp.class);
+    static boolean is_running_with_dynamic_archive = !is_dynamic_dumping;
+
+    public static void main(String args[]) throws Exception {
+        String path = args[0];
+        URL url = new File(path).toURI().toURL();
+        URL[] urls = new URL[] {url};
+        System.out.println(path);
+        System.out.println(url);
+
+        int num_loops = 1;
+        if (args.length > 1) {
+            num_loops = Integer.parseInt(args[1]);
+        }
+        loaders = new URLClassLoader[num_loops];
+        for (int i = 0; i < num_loops; i++) {
+            loaders[i] = new URLClassLoader(urls);
+        }
+
+        if (is_dynamic_dumping) {
+            // Try to load the super interfaces of CustomLoadee2 in different orders
+            for (int i = 0; i < num_loops; i++) {
+                int a = (i + 1) % num_loops;
+                loaders[a].loadClass("CustomInterface2_ia");
+            }
+            for (int i = 0; i < num_loops; i++) {
+                int a = (i + 2) % num_loops;
+                loaders[a].loadClass("CustomInterface2_ib");
+            }
+        }
+
+        for (int i = 0; i < num_loops; i++) {
+            System.out.println("============================ LOOP = " + i);
+            URLClassLoader urlClassLoader = loaders[i];
+            test(i, urlClassLoader, "CustomLoadee");
+            test(i, urlClassLoader, "CustomInterface2_ia");
+            test(i, urlClassLoader, "CustomInterface2_ib");
+            test(i, urlClassLoader, "CustomLoadee2");
+            test(i, urlClassLoader, "CustomLoadee3");
+            test(i, urlClassLoader, "CustomLoadee3Child");
+        }
+    }
+
+    private static void test(int i, URLClassLoader urlClassLoader, String name) throws Exception {
+        Class c = urlClassLoader.loadClass(name);
+        try {
+            c.newInstance(); // make sure the class is linked so it can be archived
+        } catch (Throwable t) {}
+        boolean is_shared = wb.isSharedClass(c);
+
+        System.out.println("Class = " + c + ", loaded from " + (is_shared ? "CDS" : "Jar"));
+        System.out.println("Loader = " + c.getClassLoader());
+
+        // [1] Check that CustomLoadee is defined by the correct loader
```

Comment:
The comment may not be accurate since different classes are being passed into this method.
Perhaps change it to the following?
// Check that the loaded class is defined by the correct loader